mycore-base/src/main/java/org/mycore/frontend/cli/MCRObjectCommands.java

```diff
@@ -1060,6 +1061,176 @@ public static void checkDerivatesInObjects(String baseId) {
+            // getTransformer with non-existent input successfully returns a working transformer
+            // that "successfully transforms", an error would be preferable 
+            trafo = getTransformer(transformerType);
+        }
+        MCRXMLMetadataManager mgr = MCRXMLMetadataManager.instance();
+        MCRObjectID objID = MCRObjectID.getInstance(objectID);
+        if (!mgr.exists(objID)) {
+            LOGGER.error("Object ID {} does not exist in store.", objectID);
+            return;
+        }
+        if (doValidateObjectAgainstSchema(objID, trafo)) {
+            LOGGER.info("Object {} successfully validated.", objectID);
+        } else {
+            LOGGER.error("Object {} failed its validation!", objectID);
+        }
+    }
+
+    private static boolean doValidateObjectAgainstSchema(MCRObjectID objID, Transformer trans) {
+        // MCRMetadataManager -> retrieveMCRObject() -> MCRObject.createXML already validates the contents
+        // we need to offer transformation first though, so manual talking to MCRXMLMetadataManager
+        // for the object contents, then manually using a validating XML parser later
+        MCRXMLMetadataManager mgr = MCRXMLMetadataManager.instance();
+        Document doc;
+        try {
+            doc = mgr.retrieveXML(objID);
+        } catch (IOException | JDOMException | SAXException e) {
+            LOGGER.error("Failed to retrieve object {}, unable to validate!", objID.toString());
+            e.printStackTrace();
+            return false;
+        }
+        MCRObject object = new MCRObject(doc);
+        try {
+            object.validate();
+        } catch (MCRException e) {
+            LOGGER.error("Object does not pass basic self-validation, unable to validate against schema!",
+                objID.toString());
+        }
+        String schema = object.getSchema();
+        if (schema == null) {
+            LOGGER.error("Object {} has no assigned schema, unable to validate!", objID.toString());
+            return false;
+        }
+        if (trans != null) {
+            JDOMResult res = new JDOMResult();
+            try {
+                trans.transform(new JDOMSource(doc), res);
+            } catch (TransformerException | MCRException e) {
+                LOGGER.error("Object {} could not be transformed, unable to validate!", objID.toString());
+                e.printStackTrace();
+                return false;
+            }
+            LOGGER.info("Object {} successfully transformed.", objID.toString());
+            doc = Objects.requireNonNull(res.getDocument(), "Could not get transformation result");
+        }
+        try {
+            MCRXMLParserFactory.getValidatingParser().parseXML(new MCRJDOMContent(doc));
+            return true;
+        } catch (MCRException | SAXException e) {
+            LOGGER.error("Object {} failed to parse against its schema!", objID.toString());
+            e.printStackTrace();
```

Comment:
Please use the LOGGER for stacktraces from exceptions and not e.printStackTrace()